{
  "uid" : "db1baab346829d28",
  "name" : "TC 06.01.02 - verify the card is interactive on hover",
  "fullName" : "tests.test_main_page.TestHotSellerSection#test_verify_the_card_is_interactive_on_hover",
  "historyId" : "0ae3fd0cada57213a5c961202c8dada8",
  "time" : {
    "start" : 1697334173367,
    "stop" : 1697334178712,
    "duration" : 5345
  },
  "description" : "This test checks that the card is interactive",
  "descriptionHtml" : "<p>This test checks that the card is interactive</p>\n",
  "status" : "skipped",
  "statusMessage" : "XFAIL In CI the test failed in PR#203\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55bf00bed6b3 <unknown>\n#1 0x55bf008c31e7 <unknown>\n#2 0x55bf0090a566 <unknown>\n#3 0x55bf0090a651 <unknown>\n#4 0x55bf00947a74 <unknown>\n#5 0x55bf0092c0ed <unknown>\n#6 0x55bf00945322 <unknown>\n#7 0x55bf0092be93 <unknown>\n#8 0x55bf008fe934 <unknown>\n#9 0x55bf008ff71e <unknown>\n#10 0x55bf00bb2cb8 <unknown>\n#11 0x55bf00bb6bf0 <unknown>\n#12 0x55bf00bc119c <unknown>\n#13 0x55bf00bb7808 <unknown>\n#14 0x55bf00b8427f <unknown>\n#15 0x55bf00bdbe88 <unknown>\n#16 0x55bf00bdc059 <unknown>\n#17 0x55bf00bec843 <unknown>\n#18 0x7f5c92494b43 <unknown>",
  "statusTrace" : "self = <test_main_page.TestMainPage.TestHotSellerSection object at 0x7fe0af3cbd50>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"ab03c0050551fbb3018732d2ad205f10\")>\n\n    @pytest.mark.xfail(reason=\"In CI the test failed in PR#203\")\n    @allure.title(\"TC 06.01.02 - verify the card is interactive on hover\")\n    def test_verify_the_card_is_interactive_on_hover(self, driver):\n        \"\"\"This test checks that the card is interactive\"\"\"\n        page = MainPage(driver, MAIN_PAGE_URL)\n        page.open()\n>       shadow_before, shadow_after = page.check_for_shadow_appearance_when_hovering_over_the_card()\n\ntests/test_main_page.py:23: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:21: in check_for_shadow_appearance_when_hovering_over_the_card\n    card_item = self.element_is_visible(self.locators.PRODUCT_CARD)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ab03c0050551fbb3018732d2ad205f10\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7fe0b0a7c540>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55bf00bed6b3 <unknown>\nE       #1 0x55bf008c31e7 <unknown>\nE       #2 0x55bf0090a566 <unknown>\nE       #3 0x55bf0090a651 <unknown>\nE       #4 0x55bf00947a74 <unknown>\nE       #5 0x55bf0092c0ed <unknown>\nE       #6 0x55bf00945322 <unknown>\nE       #7 0x55bf0092be93 <unknown>\nE       #8 0x55bf008fe934 <unknown>\nE       #9 0x55bf008ff71e <unknown>\nE       #10 0x55bf00bb2cb8 <unknown>\nE       #11 0x55bf00bb6bf0 <unknown>\nE       #12 0x55bf00bc119c <unknown>\nE       #13 0x55bf00bb7808 <unknown>\nE       #14 0x55bf00b8427f <unknown>\nE       #15 0x55bf00bdbe88 <unknown>\nE       #16 0x55bf00bdc059 <unknown>\nE       #17 0x55bf00bec843 <unknown>\nE       #18 0x7f5c92494b43 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1697332188525,
      "stop" : 1697332188634,
      "duration" : 109
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver",
    "time" : {
      "start" : 1697334172237,
      "stop" : 1697334173367,
      "duration" : 1130
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_result_folder",
    "time" : {
      "start" : 1697332188634,
      "stop" : 1697332188634,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "This test checks that the card is interactive",
    "status" : "skipped",
    "statusMessage" : "XFAIL In CI the test failed in PR#203\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55bf00bed6b3 <unknown>\n#1 0x55bf008c31e7 <unknown>\n#2 0x55bf0090a566 <unknown>\n#3 0x55bf0090a651 <unknown>\n#4 0x55bf00947a74 <unknown>\n#5 0x55bf0092c0ed <unknown>\n#6 0x55bf00945322 <unknown>\n#7 0x55bf0092be93 <unknown>\n#8 0x55bf008fe934 <unknown>\n#9 0x55bf008ff71e <unknown>\n#10 0x55bf00bb2cb8 <unknown>\n#11 0x55bf00bb6bf0 <unknown>\n#12 0x55bf00bc119c <unknown>\n#13 0x55bf00bb7808 <unknown>\n#14 0x55bf00b8427f <unknown>\n#15 0x55bf00bdbe88 <unknown>\n#16 0x55bf00bdc059 <unknown>\n#17 0x55bf00bec843 <unknown>\n#18 0x7f5c92494b43 <unknown>",
    "statusTrace" : "self = <test_main_page.TestMainPage.TestHotSellerSection object at 0x7fe0af3cbd50>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"ab03c0050551fbb3018732d2ad205f10\")>\n\n    @pytest.mark.xfail(reason=\"In CI the test failed in PR#203\")\n    @allure.title(\"TC 06.01.02 - verify the card is interactive on hover\")\n    def test_verify_the_card_is_interactive_on_hover(self, driver):\n        \"\"\"This test checks that the card is interactive\"\"\"\n        page = MainPage(driver, MAIN_PAGE_URL)\n        page.open()\n>       shadow_before, shadow_after = page.check_for_shadow_appearance_when_hovering_over_the_card()\n\ntests/test_main_page.py:23: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:21: in check_for_shadow_appearance_when_hovering_over_the_card\n    card_item = self.element_is_visible(self.locators.PRODUCT_CARD)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ab03c0050551fbb3018732d2ad205f10\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7fe0b0a7c540>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55bf00bed6b3 <unknown>\nE       #1 0x55bf008c31e7 <unknown>\nE       #2 0x55bf0090a566 <unknown>\nE       #3 0x55bf0090a651 <unknown>\nE       #4 0x55bf00947a74 <unknown>\nE       #5 0x55bf0092c0ed <unknown>\nE       #6 0x55bf00945322 <unknown>\nE       #7 0x55bf0092be93 <unknown>\nE       #8 0x55bf008fe934 <unknown>\nE       #9 0x55bf008ff71e <unknown>\nE       #10 0x55bf00bb2cb8 <unknown>\nE       #11 0x55bf00bb6bf0 <unknown>\nE       #12 0x55bf00bc119c <unknown>\nE       #13 0x55bf00bb7808 <unknown>\nE       #14 0x55bf00b8427f <unknown>\nE       #15 0x55bf00bdbe88 <unknown>\nE       #16 0x55bf00bdc059 <unknown>\nE       #17 0x55bf00bec843 <unknown>\nE       #18 0x7f5c92494b43 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
    "steps" : [ {
      "name" : "Open a browser",
      "time" : {
        "start" : 1697334173368,
        "stop" : 1697334173590,
        "duration" : 222
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Find a visible element",
      "time" : {
        "start" : 1697334173590,
        "stop" : 1697334178711,
        "duration" : 5121
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55bf00bed6b3 <unknown>\n#1 0x55bf008c31e7 <unknown>\n#2 0x55bf0090a566 <unknown>\n#3 0x55bf0090a651 <unknown>\n#4 0x55bf00947a74 <unknown>\n#5 0x55bf0092c0ed <unknown>\n#6 0x55bf00945322 <unknown>\n#7 0x55bf0092be93 <unknown>\n#8 0x55bf008fe934 <unknown>\n#9 0x55bf008ff71e <unknown>\n#10 0x55bf00bb2cb8 <unknown>\n#11 0x55bf00bb6bf0 <unknown>\n#12 0x55bf00bc119c <unknown>\n#13 0x55bf00bb7808 <unknown>\n#14 0x55bf00b8427f <unknown>\n#15 0x55bf00bdbe88 <unknown>\n#16 0x55bf00bdc059 <unknown>\n#17 0x55bf00bec843 <unknown>\n#18 0x7f5c92494b43 <unknown>\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
      "steps" : [ {
        "name" : "Find a present element",
        "time" : {
          "start" : 1697334173590,
          "stop" : 1697334178711,
          "duration" : 5121
        },
        "status" : "broken",
        "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55bf00bed6b3 <unknown>\n#1 0x55bf008c31e7 <unknown>\n#2 0x55bf0090a566 <unknown>\n#3 0x55bf0090a651 <unknown>\n#4 0x55bf00947a74 <unknown>\n#5 0x55bf0092c0ed <unknown>\n#6 0x55bf00945322 <unknown>\n#7 0x55bf0092be93 <unknown>\n#8 0x55bf008fe934 <unknown>\n#9 0x55bf008ff71e <unknown>\n#10 0x55bf00bb2cb8 <unknown>\n#11 0x55bf00bb6bf0 <unknown>\n#12 0x55bf00bc119c <unknown>\n#13 0x55bf00bb7808 <unknown>\n#14 0x55bf00b8427f <unknown>\n#15 0x55bf00bdbe88 <unknown>\n#16 0x55bf00bdc059 <unknown>\n#17 0x55bf00bec843 <unknown>\n#18 0x7f5c92494b43 <unknown>\n\n",
        "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.5/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
        "steps" : [ ],
        "attachments" : [ ],
        "parameters" : [ {
          "name" : "locator",
          "value" : "('css selector', \"li[class='product-item']:nth-child(6) div[class='product-item-info']\")"
        }, {
          "name" : "timeout",
          "value" : "5"
        } ],
        "stepsCount" : 0,
        "attachmentsCount" : 0,
        "shouldDisplayMessage" : true,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "locator",
        "value" : "('css selector', \"li[class='product-item']:nth-child(6) div[class='product-item-info']\")"
      }, {
        "name" : "timeout",
        "value" : "10"
      } ],
      "stepsCount" : 1,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "ad16e1cb5174e255",
      "name" : "Screenshot",
      "source" : "ad16e1cb5174e255.png",
      "type" : "image/png",
      "size" : 14924
    }, {
      "uid" : "2f9d58333ed9ad12",
      "name" : "HTML source",
      "source" : "2f9d58333ed9ad12.html",
      "type" : "text/html",
      "size" : 571
    } ],
    "parameters" : [ ],
    "stepsCount" : 3,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "driver::0",
    "time" : {
      "start" : 1697334178900,
      "stop" : 1697334178959,
      "duration" : 59
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_result_folder::0",
    "time" : {
      "start" : 1697335158583,
      "stop" : 1697335158584,
      "duration" : 1
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Main Page"
  }, {
    "name" : "feature",
    "value" : "Testing Hot Seller Section"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.xfail(reason='In CI the test failed in PR#203')"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_main_page"
  }, {
    "name" : "subSuite",
    "value" : "TestMainPage"
  }, {
    "name" : "host",
    "value" : "fv-az163-423"
  }, {
    "name" : "thread",
    "value" : "3041-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_main_page"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ ],
    "tags" : [ "@pytest.mark.xfail(reason='In CI the test failed in PR#203')" ]
  },
  "source" : "db1baab346829d28.json",
  "parameterValues" : [ ]
}